src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.GeneratedFileReplacingCompilationTracker.cs

```diff
@@ -196,6 +196,12 @@ public ValueTask<ImmutableArray<Diagnostic>> GetSourceGeneratorDiagnosticsAsync(
                 // outputs of each other.
                 return UnderlyingTracker.GetSourceGeneratorDiagnosticsAsync(solution, cancellationToken);
             }
+
+            public ValueTask<GeneratorDriverRunResult?> GetSourceGeneratorRunResultAsync(SolutionState solution, CancellationToken cancellationToken)
+            {
+                // PROTOTYPE: the run results should always be the same?
```

Comment:
Not really. The "generated file replacing" here is a specific thing we need to sometimes deal with in the IDE: if the user makes an edit, that might result in generated files being different. But if you have that generated file open, _and_ we haven't updated the text yet, we create a snapshot that says "hey, just pretend the generator gave this old state file as it's output".

So what's worrying me here a bit is this means the GeneratorDriverRunResult here being returned is technically stale here. For host outputs that's _probably_ fine, but it does mean that the returned GeneratorDriverRunResult cannot be looked at for the actual output documents, since those would be directly out of sync with what GetSourceGeneratedDocumentStatesAsync returns. I think my preference would be one of two things here:

1. Just make this method throw NotSupportedException or something similar, since I'm not sure if you'll ever need hsot outputs once you've taken this path. There's one or two other methods here where we've already done that, so it's not the end of the world.
2. Rather than this returning the GeneratorDriverRunResult, specifically make this a GetHostOutputsAsync() method, effectively moving the implementation that you have in the external access code. That way we're not exposing the definitely-out-of-sync object.